tests/lbfgs_test.py

```diff
@@ -358,6 +359,56 @@ def _run_lbfgs_solver(X, y, lmbd, n_iter):
     data = get_data()
     coef = _run_lbfgs_solver(data["X"], data["y"], 1.0, 2)
     self.assertGreater(jnp.sum(coef**2), 0)
+
+  def test_binary_logit_log_likelihood(self):
+    # See issue #409
+    rng = jax.random.PRNGKey(42)
+    N = 1000
+    beta = jnp.array([[0.5,0.5]]).T
+    income = jax.random.normal(rng, shape=(N,1))
+    x = jnp.hstack([jnp.ones((N,1)), income])
+
+    def simulate_binary_logit(x, beta):
+      beta = beta.reshape(-1,1)
+      N = x.shape[0]
+      J = beta.shape[0]
+
+      epsilon = jax.random.gumbel(rng,shape =(N,J))
+      Beta_augmented = jnp.hstack([beta, jnp.zeros_like(beta)])
+      utility = x @ Beta_augmented + epsilon
+
+      choice_idx = onp.argmax(utility, axis=1)
+      return (choice_idx).reshape(-1,1)
+
+    y = simulate_binary_logit(x, beta)
+    y = jnp.ravel(y)
+
+    # numpy version
+    def binary_logit_log_likelihood(beta, y,x):
+      lambda_xb = onp.exp(x@beta) / (1 + onp.exp(x@beta))
+      ll_i = y * onp.log(lambda_xb) + (1-y) * onp.log(1-lambda_xb)
+      ll = -onp.sum(ll_i)
+      return ll
```

Comment:
remove

Reply:
Which part do you want me to remove exactly? Having the comparison with scipy looks like a good safeguard. By the way I think we can remove the test test_first_stepsize_inf which did not reveal all issues.

Reply:
I thought it's sufficient to just check that we don't get stuck at zero (our test suite is getting longer and longer to run). But you're right, maybe it's better to compare with scipy to be entirely sure.

Reply:
Ok, I removed the previous test as it was redundant and less documented and kept the comparison to scipy.